src/zip.js

```diff
@@ -1,38 +1,49 @@
-var write = require('./write'),
-    geojson = require('./geojson'),
-    prj = require('./prj'),
-    JSZip = require('jszip');
+var write = require("./write"),
+  geojson = require("./geojson"),
+  prj = require("./prj"),
+  JSZip = require("jszip");
 
-module.exports = function(gj, options) {
+module.exports = function (gj, options, stream = false) {
+  var zip = new JSZip(),
+    layers = zip.folder(options && options.folder ? options.folder : "layers");
 
-    var zip = new JSZip(),
-        layers = zip.folder(options && options.folder ? options.folder : 'layers');
-
-    [geojson.point(gj), geojson.line(gj), geojson.polygon(gj)]
-        .forEach(function(l) {
-        if (l.geometries.length && l.geometries[0].length) {
-            write(
-                // field definitions
-                l.properties,
-                // geometry type
-                l.type,
-                // geometries
-                l.geometries,
-                function(err, files) {
-                    var fileName = options && options.types[l.type.toLowerCase()] ? options.types[l.type.toLowerCase()] : l.type;
-                    layers.file(fileName + '.shp', files.shp.buffer, { binary: true });
-                    layers.file(fileName + '.shx', files.shx.buffer, { binary: true });
-                    layers.file(fileName + '.dbf', files.dbf.buffer, { binary: true });
-                    layers.file(fileName + '.prj', prj);
-                });
+  [
+    geojson.point(gj),
+    geojson.line(gj),
+    geojson.polygon(gj),
+    geojson.multipolygon(gj),
+    geojson.multiline(gj),
+  ].forEach(function (l) {
+    if (l.geometries.length && l.geometries[0].length) {
+      write(
+        // field definitions
+        l.properties,
+        // geometry type
+        l.type,
+        // geometries
+        l.geometries,
+        function (err, files) {
+          var fileName =
+            options && options.types[l.type.toLowerCase()]
+              ? options.types[l.type.toLowerCase()]
+              : l.type;
+          layers.file(fileName + ".shp", files.shp.buffer, { binary: true });
+          layers.file(fileName + ".shx", files.shx.buffer, { binary: true });
+          layers.file(fileName + ".dbf", files.dbf.buffer, { binary: true });
+          layers.file(fileName + ".prj", prj);
         }
-    });
+      );
+    }
+  });
 
-    var generateOptions = { compression:'STORE' };
+  var generateOptions = { type: "base64", compression: "STORE" };
 
-    if (!process.browser) {
-      generateOptions.type = 'nodebuffer';
-    }
+  // Our implementation will DEFINITELY be in the browser for now.
+  // if (!process.browser) {
```

Comment:
Our use case is run in the browser, so I removed this snippet of code. I am looking for someone with a NodeJS use case to contribute the respective server side solution before this is approved. Thank you.

Reply:
Hey Charlie, please check the PR I opened (https://github.com/Agriculture-Intelligence/shp-write/pull/8) to your `master` branch that deals with this issues and others we had by abstracting more of this to the `options` parameter. It also cleans up the typing a bit. Thanks

Reply:
Actually I typed it as `Promise<any> | ReadableStream;` and that works well enough for my use case for now while adding the `ReadableStream` for you.

Reply:
I actually just added the stream bit as a copy/paste of [this contribution](https://github.com/mapbox/shp-write/pull/84). However, as I am working purely in the browser. I do not think I need that. Do with it as you please.